### PR TITLE
docs: update versioning doc

### PR DESCRIPTION
**Pull request by ocofaigh**
### Description

update versioning doc

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: docs/versioning.md [hunk 1]
```diff
@@ -1,55 +1,46 @@
-# Release compatibility
+# Release versioning
 
 Modules are updated regularly. The changes adhere to [semantic versioning](https://semver.org/), with releases labeled as `{major}.{minor}.{patch}`. [Git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) are used to identify module versions.
 
-A major release level indicates a change that is not compatible with earlier versions. A minor release includes new capabilities and includes only compatible changes. A patch includes mostly bugs fixes that are compatible with an earlier version.
+Release levels apply to both modules and submodules. For example, a major version change to a submodule results in a major release for the whole module. However, changes to examples do not drive release levels, unless the example is published as a Deployable Architecture to the IBM Cloud.
```
Comment (vburckhardt) [inline docs/versioning.md:5]: "unless the example is published as a Deployable Architecture to the IBM Cloud" -> I think we should graduate as example used as a deployable arch to a submodule to be fully consistent. It won't be possible for people to know what example are used without digging in the catalog json or in catalog ui.

File: docs/versioning.md [hunk 2]
```diff
@@ -1,55 +1,46 @@
...
+No universal criteria exist to determine terraform module versioning. However, when we evaluate changes, we consider major version updates only necessary when any of the actual infrastructure that is deployed by the module will either be destroyed or modified in a way that may affect every day operations.
 
-No universal criteria exist to determine compatibility. However, when we evaluate changes, we consider the possible assumptions that client developers might have and that indicate that a particular change is likely not compatible.
+### Major version changes
 
-### Incompatible changes
+A major version release is one that impacts the actual infrastructure that is deployed by the module itself. The following are example of major version updates:
 
-At a high level, a change that is incompatible with earlier versions is a change that causes code to stop working or might destroy infrastructure. Sometimes modules require these incompatible changes. Known incompatible changes are introduced only in major releases.
+- Adding changes that may destroy infrastructure or update resources (that may impact every day operations) that were created in the previous version
```
Comment (vburckhardt) [inline docs/versioning.md:15]: I'd split the bullet here - put the update separately along with "that may impact every day operations" that apply only to updates

File: docs/versioning.md [hunk 3]
```diff
@@ -1,55 +1,46 @@
...
+- Reducing or expanding possible values for input variables
+- Requiring a new terraform provider
+- Updating the required provider version constraints
 
-    A minor version change for new feature. A patch release for a fix.
+As some of the above types of changes could require consuming client developers to update code in order to consume the new minor version, such changes are documented in the versioning GIT release notes
 
-    To minimize disruption, the Terraform variable remains in the code as optional. A message is returned when users reference the variable to indicate that the variable is deprecated and will be removed in the next major release of the module.
-- Reducing or expanding possible values for input variables
+### Patch version changes
```
Comment (vburckhardt) [inline docs/versioning.md:30]: Missing text under this section

File: docs/versioning.md [hunk 4]
```diff
@@ -1,55 +1,46 @@
...
 
 ## Where patches are added
 
 Patches are added to the current release, regardless of the version that they are found in. In other words, if a bug is found in release `X.Y.Z` and the current release is `A.B.C`, the patch is delivered in release `A.B.(C+1)`, not in `X.Y.(Z+1)`.
 
-## Upgrading
+The following examples identify patch version changes:
```
Comment (vburckhardt) [inline docs/versioning.md:36]: This probably belongs under "Patch version changes"

**Review by vburckhardt — CHANGES_REQUESTED** [review]
Thanks for putting this together, I put some comments

File: docs/versioning.md (hunk 1 above)
Comment (ocofaigh) [reply docs/versioning.md:5]: for SLZ, the only DA left in `examples` is the quickstart one. Should we move it into `patterns`?
For PowerVS and SAP, all of their DAs come from the `examples` folder. Should we advise that they move into a different folder?

Comment (vburckhardt) [reply docs/versioning.md:5]: I'd move it out as one submodule under the vsi pattern

**Review by vburckhardt — DISMISSED** [review]

**Comment by ocofaigh** [comment]
@SirSpidey Over to you for editorial review please

**Review by SirSpidey — APPROVED** [review]
👍

**Pull request merged**